[2.5_stage] Clarify hub template RBAC

dhaiducek opened the pull request:
Addresses:
- https://github.com/stolostron/backlog/issues/24395

dhaiducek committed:
Clarify hub template RBAC

Signed-off-by: Dale Haiducek <dhaiduce@redhat.com>

oafischer commented on governance/custom_template.adoc:
```diff
@@ -468,43 +480,31 @@ See the following table for a comparison of hub cluster and managed cluster temp
+
 | Functions
-| A set of template functions that support dynamic access to Kubernetes resources and string manipulation. See <<template-functions,Template functions>> for more information.
+| A set of template functions that support dynamic access to Kubernetes resources and string manipulation. See <<template-functions,Template functions>> for more information. See the "Access control" row for lookup restrictions.
```
Checked the IBM style guide, as far as I can tell we don't highlight text references to rows within tables, so just remove both "" around Access control and you should be good. Pinging style guru @swopebe to double check.

oafischer reviewed (approved):
Looks good to me, thanks!

dockerymick committed:
Update custom_template.adoc

Added the reference for the Comparison of hub cluster and managed cluster templates table

dockerymick reviewed (approved):
Added a reference to the table section in the TOC. Thanks for pulling me in! This looks good !
